src/random_test.cpp

```diff
@@ -1607,20 +1608,63 @@ void Table::DeleteRandomRow(Thd1 *thd) {
       }
     }
   }
-
   std::string sql = "DELETE FROM " + name_ + " WHERE " +
-                    columns_->at(where)->name_ + "=" +
-                    columns_->at(where)->rand_value();
+                   columns_->at(where)->name_;
+  if (prob <= 90)
+    sql = sql + " = " + columns_->at(where)->rand_value();
+  else if (prob <= 92)
+    sql = sql + " >= " + columns_->at(where)->rand_value() + " AND " +
+         columns_->at(where)->name_ + " <= " +
+         columns_->at(where)->rand_value();
+  else if (prob <= 94)
+    sql = sql + " IN (" +
+         columns_->at(where)->rand_value() + "," +
+         columns_->at(where)->rand_value() + ")";
+  else if (prob <= 96)
+    sql = sql + " BETWEEN " +
+         columns_->at(where)->rand_value() + " AND " +
+         columns_->at(where)->rand_value();
+  else
+    sql = sql + " LIKE '%" +
+         columns_->at(where)->rand_value() + "%'";
+
   table_mutex.unlock();
   execute_sql(sql, thd);
 }
 
 void Table::SelectRandomRow(Thd1 *thd) {
   table_mutex.lock();
   auto where = rand_int(columns_->size() - 1);
+  auto prob = rand_int(102);
   std::string sql = "SELECT * FROM " + name_ + " WHERE " +
-                    columns_->at(where)->name_ + "=" +
-                    columns_->at(where)->rand_value();
+                   columns_->at(where)->name_;
+  if (prob <= 90)
+    sql = sql + " = " +
+         columns_->at(where)->rand_value();
+  else if (prob <= 92)
+    sql = sql + " >= " +
+         columns_->at(where)->rand_value();
+  else if (prob <= 94)
+    sql = sql + " >= " +
+         columns_->at(where)->rand_value() + " AND " +
+         columns_->at(where)->name_ + " <= " +
+         columns_->at(where)->rand_value();
+  else if (prob <= 96)
+    sql = sql + " IN (" +
+         columns_->at(where)->rand_value() + ", " +
+         columns_->at(where)->rand_value() + ")";
+  else if (prob <= 98)
+    sql = sql + " LIKE '%" +
+         columns_->at(where)->rand_value() + "%'";
+  else if (prob <= 100)
+    sql = sql + " BETWEEN " +
+         columns_->at(where)->rand_value() + " AND " +
+         columns_->at(where)->rand_value();
+  else
+    sql = sql + " NOT BETWEEN " +
```

Comment:
else if (rand_int(1000) = 0) .

it is more like selecting all rows. which we want very rarely.

Reply:
Ok. But IMO, rand_int(500) = 0 should be sufficient, 1/500 = 0.02%
Making it extremelly rare is as good as removing it which is not ideal